### PR TITLE
PostgreSQL and relational databases

### PR DESCRIPTION
**Pull request by Pattrigue**

### CODE REVIEW

File: sections/Chapter 3 Sprint 1/postgreSQL.tex [hunk 1]
```diff
@@ -0,0 +1,41 @@
+\subsection{PostgreSQL}
+The group last year chose to implement the database using PostgreSQL which is an relational database system \cite{knox2020}.
```
Comment (RasmusHoeyerHansen) [inline sections/Chapter 3 Sprint 1/postgreSQL.tex:2]: The database? Which one? All of them?

File: sections/Chapter 3 Sprint 1/postgreSQL.tex [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+A relational database is a collection of tables, known as relations. Relational databases are used to provide access to related data points.
```
Comment (RasmusHoeyerHansen) [inline sections/Chapter 3 Sprint 1/postgreSQL.tex:4]: I miss a mathematical describtion of relations. Something ala "A relational database can be described mathematically as a collection of relations containing tuples. Each relation contains attributes describing the domain of the values of the tuples. A relational database describes these relations in the form of a table with columns corresponding to the name and domain of the attributes of the represented relation attributes. The rows of the table correspond to tuple values of the relation.
In the context of an enterprise......"

File: sections/Chapter 3 Sprint 1/postgreSQL.tex [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+These relations consist of rows, known as tuples. Each tuple represents an entity with specific attributes represented by the columns of the given relation.
+In the context of an enterprise company, one could for example model an employee and a manager, and the relationship between them.
+The goal of the designing such relations is to represent entities as intuitively as possible, making it easy to establish relationships between them.
```
Comment (RasmusHoeyerHansen) [inline sections/Chapter 3 Sprint 1/postgreSQL.tex:7]: Is that the only goal of designing relations? It also breaks a bit away from the description of relational databases as mathematical concepts - then its immediately returned to with a description of keys.

File: sections/Chapter 3 Sprint 1/postgreSQL.tex [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+A relational database is a collection of tables, known as relations. Relational databases are used to provide access to related data points.
+These relations consist of rows, known as tuples. Each tuple represents an entity with specific attributes represented by the columns of the given relation.
+In the context of an enterprise company, one could for example model an employee and a manager, and the relationship between them.
+The goal of the designing such relations is to represent entities as intuitively as possible, making it easy to establish relationships between them. 
+
+To distinguish between tuples, keys are used to ensure that tuples in a relation can be uniquely identified.
+Foreign keys are used for modelling relationships between relations where foreign key in one relation represents the primary key of another relation.
+
+Data from a relational database can be queried using relational query languages such as SQL.
+Executing a query instructs the database system to perform a set of operations to compute a desired result, known as transactions.
```
Comment (RasmusHoeyerHansen) [inline sections/Chapter 3 Sprint 1/postgreSQL.tex:13]: Not all queries are transactions.

**Review by RasmusHoeyerHansen — CHANGES_REQUESTED** [review]
I would like a clearer separation between the mathematics of relational databases. I would also like a separation between a database design (Entity models) and SQL. 
One could consider writing something 'rød tråd' ala:
Postgres is implemented last year --> that is a relational database --> relational databases can be described using relations of tuples --> the tuples follow attributes of the relation --> to ensure unique tuples, keys are used --> In a relational database, the relations are represented in tables, with columns describing the attributes, and with each row representing a tuple of information --> to ensure good database design (minimal storage, maximum information) one can design the relations and map them to EM and implement them in database --> DDL --> DML --> transactions.
The short mention of the layers of the database is okay. It might also be worth mentioning that designing for storage and designing to represent the domain can be mutually exclusive.

**Review by Ivikhostrup — APPROVED** [review]

**Review by RasmusHoeyerHansen — APPROVED** [review]

**Pull request merged**